hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -255,40 +255,47 @@ object DefaultSource {
         Option.empty
       }
 
-      (tableType, queryType, isBootstrappedTable) match {
-        case (COPY_ON_WRITE, QUERY_TYPE_SNAPSHOT_OPT_VAL, false) |
-             (COPY_ON_WRITE, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false) |
-             (MERGE_ON_READ, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false) =>
+      val isMultipleBaseFileFormatsEnabled = metaClient.getTableConfig.isMultipleBaseFileFormatsEnabled
+      (tableType, queryType, isBootstrappedTable, isMultipleBaseFileFormatsEnabled) match {
+        case (COPY_ON_WRITE, QUERY_TYPE_SNAPSHOT_OPT_VAL, false, true) |
+             (COPY_ON_WRITE, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false, true) =>
+          new HoodieMultiFileFormatCOWRelation(sqlContext, metaClient, parameters, userSchema, globPaths)
+        case (MERGE_ON_READ, QUERY_TYPE_SNAPSHOT_OPT_VAL, false, true) |
```

Comment:
Can we handle the `isMultipleBaseFileFormatsEnabled` serarately?

Reply:
done